Xcode dependency not specified?

I got the following error while building locally following the instructions [here](https://ml-explore.github.io/mlx/build/html/install.html). 

```(base) MacBook-Pro-6:mlx enochkan$ env CMAKE_BUILD_PARALLEL_LEVEL="" pip install -e .
Obtaining file:///Users/enochkan/Desktop/cool_shit/mlx
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Installing backend dependencies ... done
  Preparing editable metadata (pyproject.toml) ... done
Building wheels for collected packages: mlx
  Building editable for mlx (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building editable for mlx (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [576 lines of output]
      running editable_wheel
      creating /private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx.egg-info
      writing /private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx.egg-info/PKG-INFO
      writing dependency_links to /private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx.egg-info/dependency_links.txt
      writing requirements to /private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx.egg-info/requires.txt
      writing top-level names to /private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx.egg-info/top_level.txt
      writing manifest file '/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx.egg-info/SOURCES.txt'
      reading manifest file '/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE'
      writing manifest file '/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx.egg-info/SOURCES.txt'
      creating '/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx-0.0.4.dev2023129+25807bb.dist-info'
      creating /private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-wheel-c3ru0dqk/.tmp-urz34er7/mlx-0.0.4.dev2023129+25807bb.dist-info/WHEEL
      running build_py
      running build_ext
      -- The CXX compiler identification is AppleClang 14.0.3.14030022
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Building MLX for arm64 processor on Darwin
      -- Building METAL sources
      -- Building with SDK for MacOS version 13.3
      
      -- Accelerate found /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/System/Library/Frameworks/Accelerate.framework
      -- Building Python bindings.
      -- Found Python: /Users/enochkan/miniconda3/bin/python3.11 (found version "3.11.5") found components: Interpreter Development Development.Module Development.Embed
      -- Performing Test HAS_FLTO
      -- Performing Test HAS_FLTO - Success
      -- Performing Test HAS_FLTO_THIN
      -- Performing Test HAS_FLTO_THIN - Success
      -- Found pybind11: /Users/enochkan/miniconda3/include (found version "2.10.4")
      -- Configuring done (1.4s)
      -- Generating done (0.0s)
      -- Build files have been written to: /private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/tmp_6ed498a.build-temp/mlx.core
      [  1%] Building unary.air
      [  2%] Building indexing.air
      [  4%] Building softmax.air
      [  5%] Building scan.air
      [  6%] Building reduce.air
      [  8%] Building sort.air
      [  9%] Building copy.air
      [ 10%] Building arange.air
      [ 12%] Building random.air
      [ 13%] Building gemv.air
      [ 15%] Building gemm.air
      [ 16%] Building conv.air
      [ 17%] Building binary.air
      [ 19%] Building arg_reduce.air
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:6:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11: note: non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/sort.metal:7:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11: note: non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:21:15: error: unknown type name 'bfloat'; did you mean 'float'?
        template <> bfloat operator()(bfloat x, bfloat y) { return fmod(x, y); }
                    ^~~~~~
                    float
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:21:33: error: unknown type name 'bfloat'; did you mean 'float'?
        template <> bfloat operator()(bfloat x, bfloat y) { return fmod(x, y); }
                                      ^~~~~~
                                      float
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:21:43: error: unknown type name 'bfloat'; did you mean 'float'?
        template <> bfloat operator()(bfloat x, bfloat y) { return fmod(x, y); }
                                                ^~~~~~
                                                float
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:21:22: error: redefinition of 'operator()'
        template <> bfloat operator()(bfloat x, bfloat y) { return fmod(x, y); }
                           ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:19:21: note: previous definition is here
        template <> float operator()(float x, float y) { return fmod(x, y); }
                          ^
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/indexing.metal:7:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/reduce.h:10:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11: noteIn file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/gemv.metal:8:
      In file included from : non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11: note: non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/softmax.metal:9:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11: note: non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/copy.metal:4:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.hIn file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/reduce.metal:7:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/reduce.h:10:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115::7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11: note: non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/scan.metal:7:
      In file included from 11: note: /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/unary.metal:6:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:: note: non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11: note: non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/arg_reduce.metal:6:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11: note: non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/random.metal:3:
      In file included from /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/utils.h:7:
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:23: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                            ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:115:11: note: non-constexpr function 'fmod' cannot be used in a constant expression
        return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
                ^
      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/metal/macos/lib/clang/31001.720/include/metal/metal_math:4553:18: note: declared here
      METAL_FUNC float fmod(float x, float y)
                       ^
      1 error generated.
      1 error generated.
      make[2]: *** [mlx/backend/metal/kernels/softmax.air] Error 1
      make[2]: *** Waiting for unfinished jobs....
      make[2]: *** [mlx/backend/metal/kernels/random.air] Error 1
      1 error generated.
      make[2]: *** [mlx/backend/metal/kernels/gemv.air] Error 1
      1 error generated.
      make[2]: *** [mlx/backend/metal/kernels/arg_reduce.air] Error 1
      1 error generated.
      1 error generated.
      make[2]: *** [mlx/backend/metal/kernels/unary.air] Error 1
      make[2]: *** [mlx/backend/metal/kernels/sort.air] Error 1
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:150:14: error: no matching function for call to object of type 'Remainder'
        c[index] = Op()(a[0], b[0]);
                   ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_ss<float, float, Remainder>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:319:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary("ss" #name #tname, itype, otype, op, ss) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:256:19: note: expanded from macro 'instantiate_binary'
        [[kernel]] void binary_op_##bopt<itype, otype, op>( \
                        ^
      <scratch space>:218:1: note: expanded from here
      binary_op_ss
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:159:14: error: no matching function for call to object of type 'Remainder'
        c[index] = Op()(a[0], b[index]);
                   ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_sv<float, float, Remainder>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:320:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary("sv" #name #tname, itype, otype, op, sv) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:256:19: note: expanded from macro 'instantiate_binary'
        [[kernel]] void binary_op_##bopt<itype, otype, op>( \
                        ^
      <scratch space>:219:1: note: expanded from here
      binary_op_sv
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:168:14: error: no matching function for call to object of type 'Remainder'
        c[index] = Op()(a[index], b[0]);
                   ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_vs<float, float, Remainder>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:321:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary("vs" #name #tname, itype, otype, op, vs) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:256:19: note: expanded from macro 'instantiate_binary'
        [[kernel]] void binary_op_##bopt<itype, otype, op>( \
                        ^
      <scratch space>:220:1: note: expanded from here
      binary_op_vs
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:177:14: error: no matching function for call to object of type 'Remainder'
        c[index] = Op()(a[index], b[index]);
                   ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_vv<float, float, Remainder>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:322:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary("vv" #name #tname, itype, otype, op, vv) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:256:19: note: expanded from macro 'instantiate_binary'
        [[kernel]] void binary_op_##bopt<itype, otype, op>( \
                        ^
      <scratch space>:221:1: note: expanded from here
      binary_op_vv
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:251:16: error: no matching function for call to object of type 'Remainder'
        c[out_idx] = Op()(a[idx.x], b[idx.y]);
                     ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_g<float, float, Remainder>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:323:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary_g("g" #name #tname, itype, otype, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:307:19: note: expanded from macro 'instantiate_binary_g'
        [[kernel]] void binary_op_g<itype, otype, op>( \
                        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:190:14: error: no matching function for call to object of type 'Remainder'
        c[index] = Op()(a[a_idx], b[b_idx]);
                   ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_g_nd1<float, float, Remainder>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:324:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary_g_nd("g" #name #tname, itype, otype, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:276:19: note: expanded from macro 'instantiate_binary_g_nd'
        [[kernel]] void binary_op_g_nd1<itype, otype, op>( \
                        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:205:16: error: no matching function for call to object of type 'Remainder'
        c[out_idx] = Op()(a[a_idx], b[b_idx]);
                     ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_g_nd2<float, float, Remainder>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:324:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary_g_nd("g" #name #tname, itype, otype, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:284:19: note: expanded from macro 'instantiate_binary_g_nd'
        [[kernel]] void binary_op_g_nd2<itype, otype, op>( \
                        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:220:16: error: no matching function for call to object of type 'Remainder'
        c[out_idx] = Op()(a[a_idx], b[b_idx]);
                     ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_g_nd3<float, float, Remainder>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:324:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary_g_nd("g" #name #tname, itype, otype, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:293:19: note: expanded from macro 'instantiate_binary_g_nd'
        [[kernel]] void binary_op_g_nd3<itype, otype, op>( \
                        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:235:16: error: no matching function for call to object of type 'Remainder'
        c[out_idx] = Op()(a[idx.x], b[idx.y]);
                     ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_g_nd<float, float, Remainder, 4>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:324:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary_g_nd("g" #name #tname, itype, otype, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:301:3: note: expanded from macro 'instantiate_binary_g_nd'
        instantiate_binary_g_dim(name, itype, otype, op, 4) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:264:19: note: expanded from macro 'instantiate_binary_g_dim'
        [[kernel]] void binary_op_g_nd<itype, otype, op, dims>( \
                        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:235:16: error: no matching function for call to object of type 'Remainder'
        c[out_idx] = Op()(a[idx.x], b[idx.y]);
                     ^~~~
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:373:1: note: in instantiation of function template specialization 'binary_op_g_nd<float, float, Remainder, 5>' requested here
      instantiate_binary_types(rem, Remainder)
      ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:342:3: note: expanded from macro 'instantiate_binary_types'
        instantiate_binary_float(name, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:328:3: note: expanded from macro 'instantiate_binary_float'
        instantiate_binary_all(name, float32, float, float, op) \
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:324:3: note: expanded from macro 'instantiate_binary_all'
        instantiate_binary_g_nd("g" #name #tname, itype, otype, op)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:302:3: note: expanded from macro 'instantiate_binary_g_nd'
        instantiate_binary_g_dim(name, itype, otype, op, 5)
        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:264:19: note: expanded from macro 'instantiate_binary_g_dim'
        [[kernel]] void binary_op_g_nd<itype, otype, op, dims>( \
                        ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:27: note: candidate template ignored: substitution failure [with T = float]
        template <typename T> T operator()(T x, T y) { return x % y; }
                                ^
      1 error generated.
      make[2]: *** [mlx/backend/metal/kernels/scan.air] Error 1
      1 error generated.
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:18:57: error: conversion from '_MLX_BFloat16' to 'complex64_t' is ambiguous
        template <typename T> T operator()(T x, T y) { return x % y; }
                                                              ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/binary.metal:150:14: note: in instantiation of function template specialization 'Remainder::operator()<_MLX_BFloat16>' requested here
        c[index] = Op()(a[0], b[0]);
                   ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/bf16.h:103:24: note: candidate function [with T = complex64_t, $1 = void]
        constexpr METAL_FUNC operator T() const thread {
                             ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:31:13: note: candidate constructor [with T = _MLX_BFloat16, $1 = void]
        constexpr complex64_t(T x) thread : real(x), imag(0) {}
                  ^
      /Users/enochkan/Desktop/cool_shit/mlx/mlx/backend/metal/kernels/complex.h:114:45: note: passing argument to parameter 'a' here
      constexpr complex64_t operator%(complex64_t a, complex64_t b) {
                                                  ^
      make[2]: *** [mlx/backend/metal/kernels/copy.air] Error 1
      16 errors generated.
      make[2]: *** [mlx/backend/metal/kernels/binary.air] Error 1
      1 error generated.
      make[2]: *** [mlx/backend/metal/kernels/reduce.air] Error 1
      1 error generated.
      make[2]: *** [mlx/backend/metal/kernels/indexing.air] Error 1
      make[1]: *** [mlx/backend/metal/kernels/CMakeFiles/mlx-metallib.dir/all] Error 2
      make: *** [all] Error 2
      Traceback (most recent call last):
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/editable_wheel.py", line 156, in run
          self._create_wheel_file(bdist_wheel)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/editable_wheel.py", line 345, in _create_wheel_file
          files, mapping = self._run_build_commands(dist_name, unpacked, lib, tmp)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/editable_wheel.py", line 268, in _run_build_commands
          self._run_build_subcommands()
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/editable_wheel.py", line 295, in _run_build_subcommands
          self.run_command(name)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "<string>", line 110, in run
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 88, in run
          _build_ext.run(self)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 345, in run
          self.build_extensions()
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 467, in build_extensions
          self._build_extensions_serial()
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 493, in _build_extensions_serial
          self.build_extension(ext)
        File "<string>", line 102, in build_extension
        File "/Users/enochkan/miniconda3/lib/python3.11/subprocess.py", line 571, in run
          raise CalledProcessError(retcode, process.args,
      subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'install']' returned non-zero exit status 2.
      /private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py:988: _DebuggingTips: Problem in editable installation.
      !!
      
              ********************************************************************************
              An error happened while installing `mlx` in editable mode.
      
              The following steps are recommended to help debug this problem:
      
              - Try to install the project normally, without using the editable mode.
                Does the error still persist?
                (If it does, try fixing the problem before attempting the editable mode).
              - If you are using binary extensions, make sure you have all OS-level
                dependencies installed (e.g. compilers, toolchains, binary libraries, ...).
              - Try the latest version of setuptools (maybe the error was already fixed).
              - If you (or your project dependencies) are using any setuptools extension
                or customization, make sure they support the editable mode.
      
              After following the steps above, if the problem still persists and
              you think this is related to how setuptools handles editable installations,
              please submit a reproducible example
              (see https://stackoverflow.com/help/minimal-reproducible-example) to:
      
                  https://github.com/pypa/setuptools/issues
      
              See https://setuptools.pypa.io/en/latest/userguide/development_mode.html for details.
              ********************************************************************************
      
      !!
        cmd_obj.run()
      Traceback (most recent call last):
        File "/Users/enochkan/miniconda3/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/enochkan/miniconda3/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/enochkan/miniconda3/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 273, in build_editable
          return hook(wheel_directory, config_settings, metadata_directory)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 436, in build_editable
          return self._build_with_temp_dir(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 389, in _build_with_temp_dir
          self.run_setup()
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 141, in <module>
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/editable_wheel.py", line 156, in run
          self._create_wheel_file(bdist_wheel)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/editable_wheel.py", line 345, in _create_wheel_file
          files, mapping = self._run_build_commands(dist_name, unpacked, lib, tmp)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/editable_wheel.py", line 268, in _run_build_commands
          self._run_build_subcommands()
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/editable_wheel.py", line 295, in _run_build_subcommands
          self.run_command(name)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "<string>", line 110, in run
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 88, in run
          _build_ext.run(self)
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 345, in run
          self.build_extensions()
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 467, in build_extensions
          self._build_extensions_serial()
        File "/private/var/folders/vx/d6lmgj8j2517s17lj_djnlvc0000gp/T/pip-build-env-m4db04at/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 493, in _build_extensions_serial
          self.build_extension(ext)
        File "<string>", line 102, in build_extension
        File "/Users/enochkan/miniconda3/lib/python3.11/subprocess.py", line 571, in run
          raise CalledProcessError(retcode, process.args,
      subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'install']' returned non-zero exit status 2.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building editable for mlx
Failed to build mlx
ERROR: Could not build wheels for mlx, which is required to install pyproject.toml-based projects
```

My friend was able to build on his m1 mac without issues. 

We both have the same versions of everything except our xcode/ clang versions are different.

Mine is Xcode 14 and his is Xcode 15.

--- a/mlx/backend/metal/kernels/binary.metal
+++ b/mlx/backend/metal/kernels/binary.metal
@@ -18,7 +18,7 @@ struct Remainder {
   template <typename T> T operator()(T x, T y) { return x % y; }
   template <> float operator()(float x, float y) { return fmod(x, y); }
   template <> half operator()(half x, half y) { return fmod(x, y); }
-  template <> bfloat operator()(bfloat x, bfloat y) { return fmod(x, y); }
+  template <> bfloat16_t operator()(bfloat16_t x, bfloat16_t y) { return fmod(x, y); }
 };
 
 struct Equal {

--- a/mlx/backend/metal/kernels/complex.h
+++ b/mlx/backend/metal/kernels/complex.h
@@ -112,5 +112,7 @@ constexpr complex64_t operator*(complex64_t a, complex64_t b) {
 }
 
 constexpr complex64_t operator%(complex64_t a, complex64_t b) {
-  return {fmod(a.real, b.real), fmod(a.imag, b.imag)};
+  auto real = a.real - (b.real * static_cast<int64_t>(a.real / b.real));
+  auto imag = a.imag - (b.imag * static_cast<int64_t>(a.imag / b.imag));
+  return {real, imag};
 }

--- a/python/mlx/nn/losses.py
+++ b/python/mlx/nn/losses.py
@@ -2,6 +2,7 @@
 
 import mlx.core as mx
 
+
 def cross_entropy(
     logits: mx.array, targets: mx.array, axis: int = -1, reduction: str = "none"
 ) -> mx.array:
@@ -24,7 +25,9 @@ def cross_entropy(
     return _reduce(loss, reduction)
 
 
-def l1_loss(predictions: mx.array, targets: mx.array, reduction: str = "none") -> mx.array:
+def l1_loss(
+    predictions: mx.array, targets: mx.array, reduction: str = "none"
+) -> mx.array:
     """
     Computes the L1 loss between predictions and targets.
 
@@ -38,11 +41,13 @@ def l1_loss(predictions: mx.array, targets: mx.array, reduction: str = "none") -
         mx.array: The computed L1 loss.
     """
     loss = mx.mean(mx.abs(predictions - targets))
-    
+
     return _reduce(loss, reduction)
 
 
-def mse_loss(predictions: mx.array, targets: mx.array, axis: int = -1, reduction: str = "none") -> mx.array:
+def mse_loss(
+    predictions: mx.array, targets: mx.array, axis: int = -1, reduction: str = "none"
+) -> mx.array:
     """
     Computes the mean squared error loss between predictions and targets.
 
@@ -57,11 +62,13 @@ def mse_loss(predictions: mx.array, targets: mx.array, axis: int = -1, reduction
         mx.array: The computed mean squared error loss.
     """
     loss = mx.mean(mx.square(predictions - targets), axis)
-    
+
     return _reduce(loss, reduction)
 
 
-def nll_loss(logits: mx.array, targets: mx.array, axis: int = -1, reduction: str = "none") -> mx.array:
+def nll_loss(
+    logits: mx.array, targets: mx.array, axis: int = -1, reduction: str = "none"
+) -> mx.array:
     """
     Computes the negative log likelihood loss between logits and targets.
 
@@ -80,7 +87,9 @@ def nll_loss(logits: mx.array, targets: mx.array, axis: int = -1, reduction: str
     return _reduce(loss, reduction)
 
 
-def kl_div_loss(logits: mx.array, targets: mx.array, axis: int = -1, reduction: str = "none") -> mx.array:
+def kl_div_loss(
+    logits: mx.array, targets: mx.array, axis: int = -1, reduction: str = "none"
+) -> mx.array:
     """
     Computes the Kullback-Leibler divergence loss between logits and targets.
 
@@ -98,7 +107,8 @@ def kl_div_loss(logits: mx.array, targets: mx.array, axis: int = -1, reduction: 
 
     return _reduce(loss, reduction)
 
-def _reduce(loss: mx.array, reduction: str = 'none'):
+
+def _reduce(loss: mx.array, reduction: str = "none"):
     if reduction == "mean":
         return mx.mean(loss)
     elif reduction == "sum":

--- a/python/tests/test_nn.py
+++ b/python/tests/test_nn.py
@@ -42,11 +42,11 @@ class TestNN(mlx_tests.MLXTestCase):
         targets = mx.array([0.5, 0.2, 0.9, 0.0])
         losses = nn.losses.l1_loss(predictions, targets, reduction="none")
         self.assertEqual(losses, 0.0)
-    
+
     def test_mse_loss(self):
         predictions = mx.array([0.5, 0.2, 0.9, 0.0])
         targets = mx.array([0.7, 0.1, 0.8, 0.2])
-        
+
         # Test with reduction 'none'
         losses_none = nn.losses.mse_loss(predictions, targets, reduction="none")
         expected_none = mx.array([0.04, 0.01, 0.01, 0.04])
@@ -61,7 +61,6 @@ class TestNN(mlx_tests.MLXTestCase):
         losses_sum = nn.losses.mse_loss(predictions, targets, reduction="sum")
         expected_sum = mx.sum(expected_none)
         self.assertEqual(losses_sum, expected_sum)
-
 
     def test_nll_loss(self):
         logits = mx.array([[0.0, -float("inf")], [-float("inf"), 0.0]])
@@ -81,7 +80,6 @@ class TestNN(mlx_tests.MLXTestCase):
         losses_sum = nn.losses.nll_loss(logits, targets, reduction="sum")
         expected_sum = mx.sum(expected_none)
         self.assertEqual(losses_sum, expected_sum)
-
 
     def test_kl_div_loss(self):
         p_logits = mx.array([[1.0, 2.0], [0.5, 1.0]])

